src/aesgcm.rs

```diff
@@ -55,35 +54,41 @@ impl AesGcmEncryptedBlock {
         })
     }
 
-    /// Return the headers Hash, NOTE you may need to merge Crypto-Key if there's
-    /// already a VAPID element present.
-    pub fn headers(self) -> HashMap<String, String> {
-        let mut result: HashMap<String, String> = HashMap::new();
+    /// Return the headers Hash.
+    /// If you're using VAPID, provide the `p256ecdsa` public key that signed the Json Web Token
+    /// so it can be included in the `Crypto-Key` field.
+    ///
+    /// Disclaimer : You will need to manually add the Authorization field for VAPID containing the JSON Web Token
+    pub fn headers(&self, vapid_public_key: Option<&[u8]>) -> Vec<(&'static str, String)> {
+        let mut result = Vec::new();
         let mut rs = "".to_owned();
-        result.insert(
-            "Crypto-Key".to_owned(),
-            format!(
-                "dh={}",
-                base64::encode_config(&self.dh, base64::URL_SAFE_NO_PAD)
+        let dh = base64::encode_config(&self.dh, base64::URL_SAFE_NO_PAD);
+        let crypto_key = match vapid_public_key {
```

Comment:
The use of an optional vapid key here makes sense to me. I'm tempted to suggest a more open-ended approach here, e.g. allowing the caller to specify arbitrary key-value pairs that get put into the header..but that's probably going to make more opportunities for things to go wrong, so special-casing this `p256ecdsa` field seems fine 👍

Reply:
The idea with this one is that the `Crypto-Key` field needs to be merged since there are several sub key-value pairs in its value (`dh` and `p256ecdsa`), one for Vapid and the other one being specific to the AESGCM scheme (this odd mix has been streamlined in AES128GCM, where the headers part is far easier to pull off). So allowing users to provide the entire key-pair would require them to effectively merge this `Crypto-Key` header themselves using the keys from their Vapid Signature and from the returned AESGCMEncryptedBlock, which I'm trying to avoid here.
Apart from this specific case, the function returns an owned `Vec` so it's pretty easy to push new key-value pairs in it if necessary, outside of the function itself.